File: src/main/java/io/github/tanguygab/tabadditions/shared/features/chat/commands/FormatCommand.java
package io.github.tanguygab.tabadditions.shared.features.chat.commands;

import io.github.tanguygab.tabadditions.shared.features.advancedconditions.AdvancedConditions;
import io.github.tanguygab.tabadditions.shared.features.chat.ChatFormat;
import lombok.Getter;
import lombok.experimental.Accessors;

@Getter
public class FormatCommand extends ChatFormat {

    @Accessors(fluent = true)
    private final boolean saveOnReload;
    private final String prefix;

    public FormatCommand(String name, String displayName, AdvancedConditions condition, AdvancedConditions viewCondition, String channel, String text, boolean saveOnReload, String prefix) {
        super(name, displayName, condition, viewCondition, channel, text.replace("%chat-format%",displayName));
        this.saveOnReload = saveOnReload;
        this.prefix = prefix;
    }

}

File: src/main/kotlin/io/github/tanguygab/tabadditions/shared/features/chat/commands/FormatCommand.kt
package io.github.tanguygab.tabadditions.shared.features.chat.commands

import io.github.tanguygab.tabadditions.shared.features.chat.ChatFormat
import me.neznamy.tab.shared.placeholders.conditions.Condition

class FormatCommand(
    name: String,
    displayName: String,
    condition: Condition?,
    viewCondition: Condition?,
    channel: String,
    text: String,
    val saveOnReload: Boolean,
    val prefix: String
) : ChatFormat(name, displayName, condition, viewCondition, channel, text.replace("%chat-format%", displayName)) 